Zend/zend_vm_def.h

```diff
@@ -9808,6 +9808,45 @@ ZEND_VM_HANDLER(209, ZEND_INIT_PARENT_PROPERTY_HOOK_CALL, CONST, UNUSED|NUM, NUM
 	ZEND_VM_NEXT_OPCODE();
 }
 
+ZEND_VM_HANDLER(210, ZEND_FETCH_DEFAULT_ARG, UNUSED|NUM, UNUSED)
+{
+	USE_OPLINE
+	SAVE_OPLINE();
+
+	zend_function *called_func = EX(call)->func;
+
+	parameter_reference pr;
+	pr.offset = opline->op1.num - 1;
+	pr.required = pr.offset < called_func->common.required_num_args;
+	pr.arg_info = &called_func->common.arg_info[pr.offset];
+	pr.fptr = called_func;
+
+	if (pr.required) {
+		zend_value_error("Cannot pass default to required parameter %u of %s%s%s()",
+			opline->op1.num,
+			called_func->common.scope ? ZSTR_VAL(called_func->common.scope->name) : "",
+			called_func->common.scope ? "::" : "",
+			ZSTR_VAL(called_func->common.function_name)
+		);
+		HANDLE_EXCEPTION();
+	}
+
+	zval default_value;
+	get_parameter_default(&default_value, &pr);
+
+	// Evaluate AST value, e.g. new class.
+	if (Z_TYPE(default_value) == IS_CONSTANT_AST) {
+		if (UNEXPECTED(zval_update_constant_ex(&default_value, called_func->common.scope) != SUCCESS)) {
+			HANDLE_EXCEPTION();
+		}
+	}
+
+	ZVAL_COPY(EX_VAR(opline->result.var), &default_value);
+	zval_ptr_dtor(&default_value);
```

Comment:
```suggestion
	ZVAL_COPY_VALUE(EX_VAR(opline->result.var), &default_value);
```

Reply:
[???](https://github.com/php/php-src/pull/15437#discussion_r1719973572)

Reply:
You should use `ZVAL_COPY()` when returning a value owned by somebody else, and `ZVAL_COPY_VALUE()` when giving away ownership. In this case, `get_parameter_default()` already increases the refcount for us, so there's no need to do it here (which is why you're getting a leak without `zval_ptr_dtor()`). I can't see the original code I commented it on, maybe I didn't look at it right.

Reply:
For posterity, I believe you can see the [original code you commented on here](https://github.com/php/php-src/pull/15437/files/fd7ac5f83b8282227235095843cb73e9d66b0717#diff-773bdb31563a0f907c75068675f6056b25f003e61f46928a31d9837ae107460d).